sw/device/tests/penetrationtests/firmware/fi/otp_fi.c

```diff
@@ -0,0 +1,320 @@
+uint32_t otp_read32_result_hw_cfg_comp[OTP_CTRL_PARAM_HW_CFG_SIZE / 4];
+uint32_t otp_read32_result_hw_cfg_fi[OTP_CTRL_PARAM_HW_CFG_SIZE / 4];
+uint32_t otp_read32_result_life_cycle_comp[OTP_CTRL_PARAM_LIFE_CYCLE_SIZE / 4];
+uint32_t otp_read32_result_life_cycle_fi[OTP_CTRL_PARAM_LIFE_CYCLE_SIZE / 4];
+
+void init_otp_mem_dump_buffers(void) {
+  for (uint32_t i = 0; i < OTP_CTRL_PARAM_VENDOR_TEST_SIZE / 4; i++) {
+    otp_read32_result_vendor_test_comp[i] = 0x00000001;
+    otp_read32_result_vendor_test_fi[i] = 0x00000001;
+  }
+  for (uint32_t i = 0; i < OTP_CTRL_PARAM_OWNER_SW_CFG_SIZE / 4; i++) {
+    otp_read32_result_owner_sw_cfg_comp[i] = 0x00000001;
+    otp_read32_result_owner_sw_cfg_fi[i] = 0x00000001;
+  }
+  for (uint32_t i = 0; i < OTP_CTRL_PARAM_HW_CFG_SIZE / 4; i++) {
+    otp_read32_result_hw_cfg_comp[i] = 0x00000001;
+    otp_read32_result_hw_cfg_fi[i] = 0x00000001;
+  }
+  for (uint32_t i = 0; i < OTP_CTRL_PARAM_LIFE_CYCLE_SIZE / 4; i++) {
+    otp_read32_result_life_cycle_comp[i] = 0x00000001;
+    otp_read32_result_life_cycle_fi[i] = 0x00000001;
+  }
+}
+
+status_t otp_vendor_test_dump(uint32_t *buffer) {
+  // Read VENDOR_TEST partition
+  TRY(otp_ctrl_testutils_dai_read32_array(&otp, kDifOtpCtrlPartitionVendorTest,
+                                          0, buffer,
+                                          OTP_CTRL_PARAM_VENDOR_TEST_SIZE / 4));
+
+  return OK_STATUS();
+}
+
+status_t otp_owner_sw_cfg_dump(uint32_t *buffer) {
+  // Read OWNER_SW_CFG partition
+  TRY(otp_ctrl_testutils_dai_read32_array(
+      &otp, kDifOtpCtrlPartitionOwnerSwCfg, 0, buffer,
+      OTP_CTRL_PARAM_OWNER_SW_CFG_SIZE / 4));
+
+  return OK_STATUS();
+}
+
+status_t otp_hw_cfg_dump(uint32_t *buffer) {
+  // Read HW_CFG partition
+  TRY(otp_ctrl_testutils_dai_read32_array(&otp, kDifOtpCtrlPartitionHwCfg, 0,
+                                          buffer,
+                                          OTP_CTRL_PARAM_HW_CFG_SIZE / 4));
+
+  return OK_STATUS();
+}
+
+status_t otp_life_cycle_dump(uint32_t *buffer) {
+  // Read LIFE_CYCLE partition
+  TRY(otp_ctrl_testutils_dai_read32_array(&otp, kDifOtpCtrlPartitionLifeCycle,
+                                          0, buffer,
+                                          OTP_CTRL_PARAM_LIFE_CYCLE_SIZE / 4));
+
+  return OK_STATUS();
+}
+
+status_t handle_otp_fi_vendor_test(ujson_t *uj) {
+  // Clear registered alerts in alert handler.
+  sca_registered_alerts_t reg_alerts = sca_get_triggered_alerts();
+
+  // Read OTP partition for comparison values
+  TRY(otp_vendor_test_dump(otp_read32_result_vendor_test_comp));
+
+  // FI code target.
+  sca_set_trigger_high();
+
+  // Point for FI
+  NOP1000;
+  NOP1000;
+  NOP1000;
+  NOP1000;
+
+  sca_set_trigger_low();
+
+  // Read OTP partition again to see if values changed
+  TRY(otp_vendor_test_dump(otp_read32_result_vendor_test_fi));
+
+  // Get registered alerts from alert handler.
+  reg_alerts = sca_get_triggered_alerts();
```

Comment:
Just a FYI - it might be useful to also register the alerts triggered inside Ibex. You could do something similar to here:
https://github.com/lowRISC/opentitan/blob/08d6cba5da6ced9d92be9a85b6c409e5a63dbd85/sw/device/tests/penetrationtests/firmware/fi/ibex_fi.c#L221-L223

Also applies to the test functions below.